Add egscomics source

haxelcaribou opened the pull request:
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

El Goonish Shive is already covered by the Hiveworks multisrc extension, but this extension is better tailored to EGS specifically. Improvents over the Hiveworks source are as follows:
- Includes EGS:NP and Sketchbook
- More descriptive comic description
- No false "Missing chapters" warning
- Has "Ongoing" status

haxelcaribou committed:
Create extension (copied from gunnerkrigg)

haxelcaribou committed:
Add page upload date

haxelcaribou committed:
Update icons

haxelcaribou committed:
Fix bug where code can fail depending on locale

haxelcaribou committed:
Add egsnp and sketchbook

AwkwardPeak7 commented on src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt:
```diff
@@ -0,0 +1,136 @@
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        val manga = arrayListOf<SManga>()
+        manga.add(
+            SManga.create().apply {
```
set `initialized = true` for all of them

AwkwardPeak7 commented on src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt:
```diff
@@ -0,0 +1,136 @@
+            chapter_number = element.previousElementSiblings().size.toFloat()
+            setUrlWithoutDomain("/" + element.attr("value"))
+            name = element.text().split(" - ", limit = 2).last()
+            date_upload = dateFormat.parse(element.text().split(" - ", limit = 2).first())?.time ?: 0L
```
```suggestion
            date_upload = dateFormat.tryParse(element.text().split(" - ", limit = 2).first())
```
need to import

AwkwardPeak7 commented on src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt:
```diff
@@ -0,0 +1,136 @@
+    override fun chapterFromElement(element: Element): SChapter {
+        return SChapter.create().apply {
+            chapter_number = element.previousElementSiblings().size.toFloat()
+            setUrlWithoutDomain("/" + element.attr("value"))
```
```suggestion
            setUrlWithoutDomain(element.absUrl("value"))
```

haxelcaribou replied:
Option values are relative to "https://www.egscomics.com/", not "https://www.egscomics.com/comic/" (or ".../egsnp/" or ".../sketchbook/"), as is assumed by jsoup. Using the absUrl function causes the "comic/" section (or "egsnp/"... you get the point) in each path to be repeated, ultimately leading to chapters not loading correctly.

AwkwardPeak7 replied:
that definitely shouldn't happen

AwkwardPeak7 commented on src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt:
```diff
@@ -0,0 +1,136 @@
+        return SChapter.create().apply {
+            chapter_number = element.previousElementSiblings().size.toFloat()
+            setUrlWithoutDomain("/" + element.attr("value"))
+            name = element.text().split(" - ", limit = 2).last()
```
they have proper titles below the `select` element, so why not use that?

haxelcaribou replied:
The links listed below the `select` element refer to story arcs, not (what by my interpretation of similar extensions would be considered as) chapters. Scraping those would miss new pages, and they only link to the first page of each arc.

AwkwardPeak7 commented on src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt:
```diff
@@ -0,0 +1,136 @@
+
+    override fun latestUpdatesRequest(page: Int): Request = throw UnsupportedOperationException()
+
+    override fun latestUpdatesSelector(): String = throw UnsupportedOperationException()
```
remove newlines between

AwkwardPeak7 commented on src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    override fun pageListParse(document: Document): List<Page> =
+        document.select("#cc-comic").mapIndexed { i, element -> Page(i, "", baseUrl + element.attr("src").substring(25)) }
```
```suggestion
        document.select("#cc-comic").mapIndexed { i, element ->
            Page(i, imageUrl = element.absUrl("src"))
        }
```

haxelcaribou committed:
Implement suggested maintainer edits

Update src/en/egscomics/src/eu/kanade/tachiyomi/extension/en/egscomics/ElGoonishShive.kt

Co-authored-by: AwkwardPeak7 <48650614+AwkwardPeak7@users.noreply.github.com>
